### PR TITLE
Revert "Remove OpenStack support"

### PR DESCRIPTION
**Pull request by MalloZup**
This reverts commit f04ef656b6933d1f073781dc5d668201494fdf9d.

### CODE REVIEW

**Comment by mbologna** [comment]
+1

File: README.md [hunk 1]
```diff
@@ -1,23 +1,20 @@
 # sumaform
 
-[![Travis CI build status](https://travis-ci.org/moio/sumaform.svg?branch=master)](https://travis-ci.org/moio/sumaform)
```
Comment (moio) [inline README.md]: These lines should not be removed

File: README.md [hunk 2]
```diff
@@ -1,23 +1,20 @@
...
 
 `sumaform` is a way to quickly configure test SUSE Manager setups with clients and servers.
 
 It got its [portmanteau](https://en.wikipedia.org/wiki/Portmanteau) name being the successor of [suminator](https://github.com/SUSE/suminator/) implemented as [Terraform](https://www.terraform.io/) modules.
 
-
 ## Installation
 
-openSUSE and SUSE Linux Enterprise Server:
+openSUSE 42.2, 42.1, SUSE Linux Enterprise Server 12 SP1:
```
Comment (moio) [inline README.md]: This line should not be changed

File: README.md [hunk 3]
```diff
@@ -1,23 +1,20 @@
...
 ```
 # Uncomment one of the following lines depending on your distro
 
-#sudo zypper addrepo https://download.opensuse.org/repositories/systemsmanagement:/sumaform/openSUSE_Leap_42.3/systemsmanagement:sumaform.repo
```
Comment (moio) [inline README.md]: Same here

File: README.md [hunk 4]
```diff
@@ -26,9 +23,9 @@ git clone https://github.com/moio/sumaform.git
 Ubuntu and Debian:
 ```
 sudo apt install alien
-wget https://download.opensuse.org/repositories/systemsmanagement:/sumaform/SLE_12_SP1/x86_64/terraform.rpm
```
Comment (moio) [inline README.md]: Same

File: README.md [hunk 5]
```diff
@@ -39,19 +36,23 @@ You will need to edit HCL ([HashiCorp Configuration Language](https://github.com
...
 `sumaform` can deploy virtual machines to:
  - single libvirt hosts
+ - OpenStack private clouds
  - Amazon Web Services
 
 The simplest, recommended setup is to use libvirt on your local host. That needs at least 8 GB of RAM in your machine.
 If you need a lot of VMs or lack hardware you probably want using an external libvirt host with bridged networking is also possible.
 
-The Amazon Web Services backend has been developed for scalability tests of SUSE Manager and it is only currently used in that context.
+The OpenStack backend is meant to be used in the SUSE engineering private SUSE OpenStack Cloud installation only. Make sure enough hardware resources are available before deploying.
```
Comment (moio) [inline README.md]: Not correct at this point, as we would target both the ECP and the Nuremberg engineering clouds. We should try to make this "plain OpenStack" as much as possible.

File: README.md [hunk 6]
```diff
@@ -61,7 +62,7 @@ Refer to the [official guides](https://www.terraform.io/docs/index.html) for a g
 For a very quick start:
 ```
 vim main.tf     # change your VM setup
-terraform init  # populates modules
```
Comment (moio) [inline README.md]: Should not be changed

File: README.md [hunk 7]
```diff
@@ -72,4 +73,4 @@ Please see [README_ADVANCED.md](README_ADVANCED.md).
 
 ### I have a problem!
 
-Check [TROUBLESHOOTING.md](TROUBLESHOOTING.md) first, if that does not help feel free to [join the Gitter chat](https://gitter.im/sumaform/Lobby) or directly drop a line to moio at suse dot de!
```
Comment (moio) [inline README.md]: Same here

File: main.tf.openstack.example [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+  description = "SCC or NCC username"
+  default = "UC7"
+}
+
+variable "cc_password" {
+  description = "SCC or NCC password"
+  default = // add password here
+}
+
+module "suma21pg" {
```
Comment (moio) [inline main.tf.openstack.example]: Please let's make this example minimal and working.

That means, please try it out at least once! :smiling_imp:

File: modules/openstack/README.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
+# OpenStack-specific configuration
+
+## Upload base images via Glance
+
+Terraform does not have support for [OpenStack Glance](http://docs.openstack.org/developer/glance/).
```
Comment (moio) [inline modules/openstack/README.md]: Is this still true today? If not, we should have a follow-up PR to upload images directly from sumaform.

File: modules/openstack/README.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+This project contains a script to upload `suminator`-specific images to Glance:
```
Comment (moio) [inline modules/openstack/README.md]: Suminator is dead! Long live Sumaform!

**Comment by MalloZup** [comment]
@moio i thought that you wanted a revert PR + another PR with the changes. :japanese_ogre:  i will do in one PR ok for me :dog2:

File: README.md (hunk 4 above)
Comment (MalloZup) [reply README.md]: updated with sp3

File: main.tf.openstack.example (hunk 8 above)
Comment (MalloZup) [reply main.tf.openstack.example]: ok 👿 🤖 🚀

File: modules/openstack/README.md [hunk 11]
```diff
@@ -0,0 +1,57 @@
+# OpenStack-specific configuration
+
+## Upload base images via terraform into the cloud
+
+Terraform have support for [OpenStack Glance](https//www.terraform.io/docs/providers/openstack/r/images_image_v2.html).
+
+### Manual upload via glance script.
```
Comment (MalloZup) [inline modules/openstack/README.md]: this section can be removed

File: modules/openstack/host/main.tf [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+  name = "${var.name}"
+  image_name = "${var.image}"
+  flavor_name = "xxl-sumaform"
+  security_groups = ["default"]
+  region = ""
+  count = "${var.count}"
+
+  network {
+    access_network = true
+    name = "fixed"
```
Comment (MalloZup) [inline modules/openstack/host/main.tf]: need to create a variable ´´´openstack_network_name´´´

**Comment by MalloZup** [comment]
ok @moio  i pushed the code until now that i use.

With this code we can do mostly everything, however we need to have the permission on the cloud to change the flavor.

I need also to create a variable for openstack_network

File: modules/openstack/host/main.tf (hunk 12 above)
Comment (moio) [reply modules/openstack/host/main.tf]: just-do-it-hed-2013

Comment (MalloZup) [reply modules/openstack/host/main.tf]: i didn't have the time, was a comment for me 🤣

**Comment by MalloZup** [comment]
closing in favor of the latest PR

**Comment by moio** [comment]
Closed in favour of #266 which includes big chunks of the work here. Big thanks to @MalloZup for getting this started - I will also credit you when the PR is merged!

**Pull request closed without merging**